### ISSUE TITLE
what is "must know" and what is an "asp net developer"?

### ISSUE THREAD

**Issue by rasoulabbasi20n** [MoienTajik/AspNetCore-Developer-Roadmap#157]
Must know means you have to know this to become a blah blah person. So if you think like this, you will answer these questions differently:

- Do I have to know about Serilog/Caching/Inheritance hierarchy types in ORMs/Microservices(!)/MediatR  to be an Asp.net developer?

Is there any difference between an Asp Net developer, a Senior Asp Net developer, and a backend developer?

**Comment by MoienTajik** [comment]
> Do I have to know about Serilog/Caching/Inheritance hierarchy types in ORMs/Microservices(!)/MediatR to be an Asp.net developer?

Yes, understanding Serilog, Caching, inheritance hierarchy types in ORMs, Microservices, and MediatR is valuable for an ASP.NET developer. These are not just **theoretical** concepts, but are **practical** tools and patterns widely adopted in the industry. For instance, a significant number of projects leverage Serilog for logging due to its flexibility and rich feature set. Additionally, if you’re working with Entity Framework as your ORM, it's crucial to grasp the different inheritance types it supports, as this knowledge directly influences how you model and manage your entities.

The essence of this roadmap is to reflect the current landscape of tools, libraries, architectures, etc., that are prevalent in the industry. This approach ensures that you're not just learning theoretically but are also aligning your skills with what is expected and required in real-world scenarios by various companies.

> Is there any difference between an Asp Net developer, a Senior Asp Net developer, and a backend developer?

I believe your question diverges from the primary intent and context of this repository, which is the ASP.NET Core Developer Roadmap 2024. The roadmap is designed to provide guidance and resources for mastering ASP.NET Core specifically, rather than delineating the nuances between different job titles or roles in the broader field of software development.

**Comment by rasoulabbasi20n** [comment]
Well, I get your point. But many of these are not related to Mastering ASP .NET Core. If you want to master ASP .NET Core, why would you need to know about microservices? That's entirely another matter. Mastering ASP .NET Core means that you have to master the framework, not the handy tools that are being used broadly in projects. ASP Net Core concepts are about how it processes a request and what interfaces it exposes for developers to handle it. Your selection of tools and libraries can be a good reference, but it's not necessary to become a master of ASP Net Core. In the best case, they belong to the "good to know" category.

**Comment by MoienTajik** [comment]
> If you want to master ASP .NET Core, why would you need to know about microservices? That's entirely another matter. Mastering ASP .NET Core means that you have to master the framework, not the handy tools that are being used broadly in projects.

Understanding the entire ecosystem is the focus of this roadmap. Following your suggestion would necessitate creating over +10 separate roadmaps per tool/library! Additionally, if you explore the roadmaps for other languages, you'll notice a similar pattern. In real-world projects, you're likely to encounter all these elements, and in my view, as a senior developer, mastering all these aspects is essential eventually.
